zion/src/test/scala/edu/uci/ics/cloudberry/zion/model/impl/JSONParserTest.scala

```diff
@@ -34,5 +35,36 @@ class JSONParserTest extends Specification with TestQuery {
       val expectQuery = Query(schema.dataset, Seq.empty, filter, Seq.empty, None, Some(selectRecent))
       actualQuery must_== expectQuery
     }
+    "parse long values " in {
```

Comment:
Yes for the types. 
About the bags, tricky question. I can't think in a usage for bags as values right now.